control_apps/src/main.rs

```diff
@@ -0,0 +1,409 @@
+        if self.is_locked() {
+            return "locked";
+        } else {
+            return "unlocked";
+        }
+    }
+}
+
+impl ControlMngr {
+    fn init(
+        valve_vent: Valve,
+        gains_vent: PIDgains,
+        valve_dump: Valve,
+        gains_dump: PIDgains,
+    ) -> Self {
+        info!("Initializing Control Manager...");
+        info!("\tvent valve {:}", valve_vent);
+        info!("\tdump valve {:}", valve_dump);
+        info!("\ttarget alt {:}", DEFAULT_TARGET_ALTITUDE);
+        return ControlMngr {
+            mode: ControlMode::Init,
+            valve_vent,
+            valve_dump,
+            target_altitude: DEFAULT_TARGET_ALTITUDE, // bogus target altitude
+            altitude_error: 0.0,                      // bogus altitude error
+            gains_vent,
+            gains_dump,
+        };
+    }
+    fn safe(&mut self) {
+        // running but not allowed to actuate valves
+        let previous_mode = self.mode;
+        // unlock the valves if they are locked to force the next step!
+        self.valve_vent.unlock();
+        self.valve_dump.unlock();
+        // execute mode transition
+        self.mode = ControlMode::Safe;
+        warn!("CtrlMode transition: {:} --> {:}", previous_mode, self.mode);
+        // lock the vent valve closed
+        self.valve_vent.set_pwm(0.0);
+        self.valve_vent.lock();
+        // lock the dump valve closed
+        self.valve_dump.set_pwm(0.0);
+        self.valve_dump.lock();
+    }
+    fn abort(&mut self) {
+        // dump all ballast and lock balloon valve closed
+        let previous_mode = self.mode;
+        // unlock the valves if they are locked to force the next step!
+        self.valve_vent.unlock();
+        self.valve_dump.unlock();
+        // execute mode transition
+        self.mode = ControlMode::Abort;
+        warn!("CtrlMode transition: {:} --> {:}", previous_mode, self.mode);
+        // lock the vent valve closed
+        self.valve_vent.set_pwm(0.0);
+        self.valve_vent.lock();
+        // lock the dump valve open
+        self.valve_dump.set_pwm(1.0);
+        self.valve_dump.lock();
+    }
+    fn idle(&mut self) {
+        // enter mode for holding altitude, nothing to do
+        let previous_mode = self.mode;
+        if previous_mode != ControlMode::Safe
+            && previous_mode != ControlMode::Abort
+            && previous_mode != ControlMode::Init
+        {
+            // unlock the valves
+            self.valve_vent.unlock();
+            self.valve_dump.unlock();
+            self.mode = ControlMode::Idle;
+            info!("CtrlMode transition: {:} --> {:}", previous_mode, self.mode);
+        } else {
+            warn!(
+                "CtrlMode transition ({:} --> {:}) not allowed! Ignoring...",
+                previous_mode,
+                ControlMode::Idle
+            );
+        }
+    }
+    fn vent(&mut self) {
```

Comment:
```suggestion
    }
    
    fn vent(&mut self) {
```
Style nit for legibility: give a space between the previous function and next function.